Added Provenance resource to HapiFhirLabOrderConverter

tjohnson7021 opened the pull request:
# Added Provenance resource to HapiFhirLabOrderConverter

- added Provenance Resource to implementation so that it can accept and convert that part of the resource bundle
- added a test for bundle timestamps
## Issue #84

Commit:
Added Provenance resource to HapiFhir converter

- added Provenance Resource fields to implementation

Commit:
Added logger statement

- added logger statement for provenance resource when sending an order

Commit:
Refactored order datetime

- refactored order dateTime so that converted bundle would reflect all the same times

halprin commented on app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiLabOrderConverter.java:
```diff
@@ -167,8 +170,24 @@ private ServiceRequest createServiceRequest(final Patient patient) {
+        provenance.setRecorded(orderDate);
+        provenance.setActivity(
+                new CodeableConcept(
+                        new Coding(
+                                "http://terminology.hl7.org/CodeSystem/v2-0003",
+                                "021",
+                                "OML - Laboratory order")));
```
I see some duplication from the `createMessageHeader` method and this code.  I bet this `Coding` could be created once and used in both places.

tjohnson7021 replied:
This should be refactored appropriately now... @halprin

halprin commented on app/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiLabOrderConverter.java:
```diff
@@ -44,21 +45,23 @@ private HapiLabOrderConverter() {}
         var patient = createPatientResource(demographics);
-        var serviceRequest = createServiceRequest(patient);
+        var serviceRequest = createServiceRequest(patient, orderDateTime);
         var messageHeader = createMessageHeader();
```
Just checking, will you be adding the `meta` field to the `MessageHeader` resource in this PR or a future PR?  That was also a new addition to to the `lab_order.json`.

Commit:
Added unit test for Provenance resource additions

- added test to check that timestamps match for certain resources in the bundle
- Provenance.recorded
- ServiceRequest.authoredOn
- Bundle timestamp

Commit:
Refactored duplicated code

- refactored omlLabOrderCoding in createMessageHeader and createProvenanceResource

Commit:
Merge branch 'main' into task/84-add_provenance_resource

Commit:
Merge branch 'main' into task/84-add_provenance_resource

Commit:
Fixed Meta class import

Commit:
Merge branch 'main' into task/84-add_provenance_resource

Commit:
Fix linting issue
